### PR TITLE
Group4: Task 3

### PR DESCRIPTION
**Pull request by corebonts**
1. Only one level of nesting
    2. No usage of else
    5. No acronyms

### CODE REVIEW

File: src/main/java/de/cas/mse/exercise/csv/CsvReader.java [hunk 1]
```diff
@@ -1,19 +1,48 @@
...
+        this.csvUi = csvUi;
+    }
+
+    @Override
+    public List<List<String>> filter(int col, String word) {
+        return data.stream().filter(row -> filterApplies(row, col, word)).collect(Collectors.toList());
+    }
+
+    private boolean filterApplies(List<String> row, int col, String word) {
+        return row.get(col).matches(String.format(".*\\b%s\\b.*", word));
```
Comment (zopad) [inline src/main/java/de/cas/mse/exercise/csv/CsvReader.java]: Why is this magic needed? Isn't row.get(col).contains(word) enough?

Comment (corebonts) [reply src/main/java/de/cas/mse/exercise/csv/CsvReader.java]: We considered "word" as a real (whole) word.

File: src/main/java/de/cas/mse/exercise/csv/CsvReader.java [hunk 2]
```diff
@@ -12,37 +12,48 @@
...
-        return data.stream().filter(row -> filterApplies(row, col, word)).collect(Collectors.toList());
+    public List<List<String>> filter(int columnIndexForFiltering, String wordToContain) {
+        String filterRegex = String.format(".*\\b%s\\b.*", wordToContain);
+        return rows.stream()
+            .filter(row -> filterApplies(row, columnIndexForFiltering, filterRegex))
+            .collect(Collectors.toList());
+    }
+
+    private List<List<String>> getRows(List<String> allLines) {
+        // First line contains the header
```
Comment (zopad) [inline src/main/java/de/cas/mse/exercise/csv/CsvReader.java:43]: Could it be that the first line is already skipped from the input? This seems unrelated to the getRows method

File: src/main/java/de/cas/mse/exercise/csv/CsvReader.java [hunk 3]
```diff
@@ -12,37 +12,48 @@
...
+        return allLines.stream().skip(1).map(this::parseLine).collect(Collectors.toList());
     }
 
-    private boolean filterApplies(List<String> row, int col, String word) {
-        return row.get(col).matches(String.format(".*\\b%s\\b.*", word));
+    private List<String> parseLine(String line) {
```
Comment (zopad) [inline src/main/java/de/cas/mse/exercise/csv/CsvReader.java:47]: parseLine is very generic, e.g. getCells could be better?

File: src/main/java/de/cas/mse/exercise/csv/CsvReader.java (hunk 1 above)
Comment (zopad) [reply src/main/java/de/cas/mse/exercise/csv/CsvReader.java]: Not sure if I understand. You can still search for phrases like "Govcaz Mill" with contains.

Comment (corebonts) [reply src/main/java/de/cas/mse/exercise/csv/CsvReader.java]: But searching for "Ka" would not return "Kaka manó", only if you search for "Kaka"

File: src/main/java/de/cas/mse/exercise/csv/CsvReader.java [hunk 4]
```diff
@@ -1,19 +1,59 @@
...
-	public void run(final File csvFile) throws Exception {
-		// TODO: Start coding here
-	}
-	
-	public void setCsvUi(final CsvUi csvUi) {
-		this.csvUi = csvUi;
-	}
+    private CsvUi csvUi;
+    private List<String> headers;
+    private List<List<String>> rows;
```
Comment (TehMillhouse) [inline src/main/java/de/cas/mse/exercise/csv/CsvReader.java:16]: Could be encapsulated in its own class

File: src/main/java/de/cas/mse/exercise/csv/CsvReader.java (hunk 2 above)
Comment (corebonts) [reply src/main/java/de/cas/mse/exercise/csv/CsvReader.java:43]: We did not want to make an additional copy of the list, nor passing Stream<> around.

File: src/main/java/de/cas/mse/exercise/csv/CsvReader.java (hunk 3 above)
Comment (corebonts) [reply src/main/java/de/cas/mse/exercise/csv/CsvReader.java:47]: We're in a CsvReader, what else would it be? :D

File: src/main/java/de/cas/mse/exercise/csv/CsvReader.java [hunk 5]
```diff
@@ -1,19 +1,59 @@
...
+        render();
+    }
+
+    public void setCsvUi(final CsvUi csvUi) {
+        this.csvUi = csvUi;
+    }
+
+    @Override
+    public List<List<String>> filter(int columnIndexForFiltering, String wordToContain) {
+        String filterRegex = String.format(".*\\b%s\\b.*", wordToContain);
```
Comment (TehMillhouse) [inline src/main/java/de/cas/mse/exercise/csv/CsvReader.java:36]: See `String.contains`?

File: src/main/java/de/cas/mse/exercise/csv/CsvReader.java (hunk 1 above)
Comment (zopad) [reply src/main/java/de/cas/mse/exercise/csv/CsvReader.java]: Ah, you're right

Comment (zopad) [reply src/main/java/de/cas/mse/exercise/csv/CsvReader.java]: If that's the expected behaviour.. :D

File: src/main/java/de/cas/mse/exercise/csv/CsvReader.java (hunk 4 above)
Comment (corebonts) [reply src/main/java/de/cas/mse/exercise/csv/CsvReader.java:16]: yes. could be

File: src/main/java/de/cas/mse/exercise/csv/CsvReader.java (hunk 5 above)
Comment (corebonts) [reply src/main/java/de/cas/mse/exercise/csv/CsvReader.java:36]: See our conversation with @zopad
